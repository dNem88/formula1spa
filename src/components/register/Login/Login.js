import React, { useState,  useContext } from 'react';
import userContext from '../../../context/userContext'
import {useNavigate, Link} from 'react-router-dom'
import styles from './login.module.css'
import Input from '../../common/input/Input'
import Form from '../../common/form/Form'
import SubmitButton from '../../common/submitButton/SubmitButton';
import FormRedirect from '../../common/formRedirect/FormRedirect'

function Login() {
    let context = useContext(userContext);
    const navigate = useNavigate()

    const [formdata, setFormdata] = useState({
        username: '',
        password: '',
        errorMessage: ''
    })
    const submitBtn = React.createRef()

    function changeHandler(e) {
        setFormdata({
            ...formdata, [e.target.id]: e.target.value, errorMessage: ''
        })
    }
    async function submitHandler(e) {
        e.preventDefault();
        submitBtn.current.disabled = true;
        try {
            let response = await context.login(formdata)
            if (response.message) {
                throw new Error(response.message)
            } 
            if (response._id && response.authToken) {
                navigate('/')
            }
        } catch(err) {
            submitBtn.current.disabled = false;
            setFormdata({
                ...formdata,
                errorMessage: err.message
            })
        }
    }
    
    return (
        <Form submitHandler={submitHandler} heading={'SIGN IN'}>
            <Input type={'text'} onChange={changeHandler} labelContent={'Username'} id={'username'} value={formdata.username}/>
            <Input type={'password'} onChange={changeHandler} labelContent={'Password'} id={'password'} value={formdata.password}/>
            <SubmitButton text={'SIGN IN'} ref={submitBtn}/>
            <p className={styles.error}>{formdata.errorMessage}</p>
            <FormRedirect linkText={'Register with F1'} label={"Dont't have an account yet?"} href={'/auth/register'}/>
        </Form>
    )
}

export default Login