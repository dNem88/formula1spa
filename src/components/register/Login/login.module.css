
.error{
    color: var(--color-red);
    padding: 0;
}
