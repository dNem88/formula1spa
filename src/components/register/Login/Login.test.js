import renderer from 'react-test-renderer';
import Login from './Login'
import React from 'react'
import userContext from '../../../context/userContext'
import {
    MemoryRouter as Router
} from 'react-router-dom'
describe('test  Login component', () => {
  it('Should render Login form', () => {
    let loginForm = renderer.create(
        <Router>
            <userContext.Provider value={{isLogged: false}}>
                <Login />
            </userContext.Provider>
        </Router>
    ).toJSON()
    expect(loginForm.children[0].children[0].type).toBe('h2')
    expect(loginForm.children[0].children[0].children[0]).toEqual('SIGN IN')
    const loginComponent = renderer.create(
        <Router>
            <userContext.Provider value={{isLogged: false}}>
                <Login />
            </userContext.Provider>
        </Router>
    )
    const instance = loginComponent.root;
    let inputElements = instance.findAllByType('input')
    expect(inputElements.length).toBe(2)
    expect(loginForm).toMatchSnapshot()
    expect(loginForm.type).toBe('form')
    let usernameLabel1 = loginForm.children[1]
    let usernameInput1 = loginForm.children[2]
    expect(usernameLabel1.props.htmlFor).toEqual(usernameInput1.props.id)
    expect(usernameInput1.type).toBe('input')
    expect(usernameLabel1.type).toBe('label')
    let passwordLabel2 = loginForm.children[3]
    let passwordInput2 = loginForm.children[4]
    expect(passwordLabel2.props.htmlFor).toEqual(passwordInput2.props.id)
    expect(passwordInput2.type).toBe('input')
    expect(passwordLabel2.type).toBe('label')
  })
  
  
})