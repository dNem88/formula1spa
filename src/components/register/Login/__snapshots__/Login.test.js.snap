// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Login component Should render Login form 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <h2>
      SIGN IN
    </h2>
  </div>
  <label
    htmlFor="username"
  >
    Username
  </label>
  <input
    id="username"
    onChange={[Function]}
    placeholder="username"
    type="text"
    value=""
  />
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    id="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <button
    type="submit"
  >
    SIGN IN
  </button>
  <p
    className="error"
  >
    
  </p>
  <div
    className="link-container"
  >
    <p>
      Dont't have an account yet?
    </p>
    <a
      href="/auth/register"
      onClick={[Function]}
    >
      Register with F1
    </a>
  </div>
</form>
`;
